src/clr-angular/data/stack-view/stack-block.ts

```diff
@@ -84,16 +89,19 @@ export class ClrStackBlock implements OnInit {
     @SkipSelf()
     @Optional()
     private parent: ClrStackBlock,
+    public stackBlockService: StackBlockService,
+    @Inject(UNIQUE_ID) uniqueId: string,
     public commonStrings: ClrCommonStringsService
   ) {
     if (parent) {
       parent.addChild();
     }
+    this.stackBlockService.id = uniqueId;
```

Comment:
Shouldn't we use the same concept as we used in our tooltip component? First, we need to check user-defined id before setting out own unique id?

Reply:
This sets the unique ID, and if the user sets one the input setter overrides it.

Reply:
Do we have the input setter for id in this component? We are setting the id only here in the constructor?

Reply:
Look at the template and notice that the id is put into the `dt` element which is internalized. Therefore an ID provided on the stack block should not be moved into the child element.

Reply:
Actually, I don't think we don't need to concern about user-defined ids. As the `dt` element is hidden in the stackblock's template.